### DISCUSSION TITLE
The target script file "x" must not call tar_make()

### DISCUSSION THREAD

**Discussion by johanneskoch94** [ropensci/targets#688] [Help] [answered]
Hi everyone, 
i just got started using targets (amazing package!) and I have stumbled upon an error, which I hope someone could help me with :) 
This is the error message that I get, when I try to reference a targets pipeline in another. 

"The target script file "x" must not call tar_make() or similar functions that would source the target script again and cause infinite recursion." 
It's thrown by the function `tar_assert_script`. 

I was wondering if what I'm trying to do shouldn't be done, or if this is assertion should be specified so that it makes sure that the call is actually recursive. Because I'm trying to call a _different_ target script.
For illustration's sake, I have here some code. I thought at first maybe that only `tar_make()` was the issue, but as the error message indicates, similar functions also cause the issue:

```
if (!purrr::is_empty(tar_outdated(store = "other_store", script ="other_script"))) {
  stop("The targets pipeline 'other' has outdated targets. Please run that pipeline before running this one again.")
}
```

**Comment by wlandau** [comment, marked as answer]
I don't see anything wrong with what you are doing, that error message is just meant to catch infinite recursion on the same data store. To work around it, I would recommend invoking `tar_outdated()` in a wrapper script instead of `_targets.R`. But if you still want to work inside `_targets.R`, one alternative is to call `eval(parse(text = "tar_outdated(store = \"other_store\", script = \"other_script\")"))`, which fools `all.vars(functions = TRUE)` in [`tar_assert_script()`](https://github.com/ropensci/targets/blob/47db06047669c779f8bfcaab77fdc7af5979e28d/R/utils_assert.R#L478-L526).

**Reply by johanneskoch94** [reply]
Thanks a lot for the quick and helpful reply!
